raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1490,50 +1390,36 @@ private FetchRequestData buildFetchRequest() {
             .setReplicaId(quorum.localId);
     }
 
-    private void maybeSendFetch(long currentTimeMs, int leaderId) throws IOException {
+    private void maybeSendLeaderFetch(long currentTimeMs, int leaderId) throws IOException {
         maybeSendRequest(currentTimeMs, leaderId, this::buildFetchRequest);
     }
 
-    private FindQuorumRequestData buildFindQuorumRequest() {
-        return new FindQuorumRequestData()
-            .setReplicaId(quorum.localId)
-            .setHost(advertisedListener.getHostString())
-            .setPort(advertisedListener.getPort())
-            .setBootTimestamp(bootTimestamp);
-    }
+    private void maybeSendVoterFetch(long currentTimeMs) throws IOException {
+        // TODO: After a fetch timeout, do we want to avoid sending the request
```

Comment:
I'd say yes, but that can be done as a state transition as in that new state we would not timer on the fetching.

Reply:
I am not so sure. It is possible that all voters go away at the same time (e.g. during a network partition). I think we have to be able to try them all in order to recover from all cases. I will remove this TODO.